exchange/src/exchange/tickers/engine/order_container.hpp

```diff
@@ -27,61 +30,18 @@ class OrderBook {
     std::unordered_map<double, double> ask_levels_;
 
 public:
-    /**
-     * @brief Get the price->quantity map for a SIDE
-     */
-    const std::unordered_map<double, double>&
-    get_levels(util::Side side) const
-    {
-        return side == util::Side::buy ? bid_levels_ : ask_levels_;
-    }
+    OrderBook(std::shared_ptr<LevelUpdateGenerator> level_update_generator = nullptr) :
+        level_update_generator_(level_update_generator)
+    {}
```

Comment:
you should rarely use default parameters. Overloading is almost always a better solution

```suggestion
    explicit OrderBook() : level_update_generator_(nullptr) {}

    explicit OrderBook(std::shared_ptr<LevelUpdateGenerator> level_update_generator) :
        level_update_generator_(level_update_generator)
    {}
```

Reply:
subjective, i prefer default

Reply:
i'm not being subjective here, default arguments tend to lead to issues with overloading

https://quuxplusone.github.io/blog/2020/04/18/default-function-arguments-are-the-devil/

Reply:
@stevenewald  [https://stackoverflow.com/a/24171267](url) 
I prefer overloading in this case especially the default constructor for orderbook is used for testing purposes, it is good to separate them

Reply:
Ok sure let's go with overloading